covid19_npis/model/number_of_tests.py

```diff
@@ -15,10 +16,85 @@
+    new cases are simply multiplied by the weekend factor on the days set by ``weekend_days``
+    The weekend factor :math:`f_w` follows a Lognormal distribution with
+    median ``pr_mean_weekend_factor`` and sigma ``pr_sigma_weekend_factor``. It is hierarchically constructed if
+    the input is two-dimensional by the function :func:`hierarchical_normal` with default arguments.
+    The offset from Sunday :math:`\Phi_w` follows a flat :class:`~pymc3.distributions.continuous.VonMises` distribution
+    and is the same for all regions.
+    Parameters
+    ----------
+    cases : :class:`~theano.tensor.TensorVariable`
+        The input array of daily new cases, can be one- or two-dimensional
+    name_cases : str or None,
+        The name under which to save the cases as a trace variable.
+        Default: None, cases are not stored in the trace.
+    week_modulation_type : str
+        The type of modulation, accepts ``"step"`` or  ``"abs_sine`` (the default).
+    pr_mean_weekend_factor : float
+        Sets the prior mean of the factor :math:`f_w` by which weekends are counted.
+    pr_sigma_weekend_factor : float
+        Sets the prior sigma of the factor :math:`f_w` by which weekends are counted.
+    weekend_days : tuple of ints
+        The days counted as weekend if ``week_modulation_type`` is ``"step"``
+    model : :class:`Cov19Model`
+        if none, it is retrieved from the context
+    Returns
+    -------
+    new_cases : :class:`~theano.tensor.TensorVariable`
+    """
+
+    log.debug("Week modulation")
+
+    """
+    # TODO: - general check of function
+            - which shape does the modulation have to be? i guess time,country,age?
+            - check prior parameters
+            - check: which ones should be modulated? (mostly: positive tests!)
+            - different modulations across: age, country?
+            - check: are (cumulative) case numbers same as in unmodulated case? need some kind of normalization?
+            - store and plot parameters at end
+    """
+    # offset-distribution of weekly modulation minimum
+    offset = yield VonMises(
+        name=name + "_offset",
+        loc=0,
+        concentration=1,
+        conditionally_independent=True,
+        event_stack=(1, modelParams.num_countries, 1),
+        shape_label=(None, "country", None),
+    )
+
+    # amplitude of weekly modulation
+    weight_cross = yield Normal(
+        name=name + "_weight",
+        loc=0,
+        scale=2,
+        conditionally_independent=True,
+        event_stack=(1, modelParams.num_countries, 1),
+        shape_label=(None, "country", None),
+        # transform=transformations.SoftPlus(scale=0.5),
+    )
+    weight = tf.math.sigmoid(weight_cross)
+
+    t = pd.date_range(start=modelParams.data_begin,end=modelParams.data_end).weekday    # create array with weekdays
+    f = (1 - weight) * (1 - tf.math.abs(tf.math.sin(t[np.newaxis,...,np.newaxis,np.newaxis] / 7 * np.pi + offset / 2))) # modulation factor
```

Comment:
tf.newaxis != np.newaxis not sure if it makes a difference tho

Reply:
changed accordingly. tf.pi does not exist, tf.constant(np.pi) seems to be the way to go.
i'm not entirely happy with the tf.newaxis part - there should be a more elegant way to broadcast it, but i couldnt get it done with tfs broadcast_to-functionality